### ISSUE TITLE
Incremental processing with `@Configuration` not working correctly

### ISSUE THREAD

**Issue by torbsto** [micronaut-projects/micronaut-core#5679]
### Steps to Reproduce

1. Clone the repository: https://github.com/torbsto/micronaut-issue.
1. Go into the project's subdirectory: `cd ./micronaut-issue/incremental-gradle-compilation`.
1. Run `./gradlew compileJava`: The build should be successful.
1. Change something in the `com.example.test` package, e.g., `sed -i '/1/s//2/' src/main/java/com/example/test/TestSingleton.java`.
1. Run `./gradlew compileJava` again. This throws an exception ([full stacktrace](https://github.com/torbsto/micronaut-issue/blob/main/incremental-gradle-compilation/stacktrace.txt)).
1. Running `./gradlew compileJava` another time doesn't throw an exception.

### Environment Information

- **Operating System**: Ubuntu 20.4 in WSL2
- **Micronaut Version:** 2.5.7
- **JDK Version:** OpenJDK 11

### Description

It seems [`@Configuration`](https://github.com/torbsto/micronaut-issue/blob/6dd53fd2b1693efb10ddaadcd90c34672fda6fcd/incremental-gradle-compilation/src/main/java/com/example/test/package-info.java#L1) doesn't work with Gradle's incremental compilation. Whenever I change source code located in the annotated package (`com.example.test`), the compilation task throws an exception ([full stacktrace](https://github.com/torbsto/micronaut-issue/blob/main/incremental-gradle-compilation/stacktrace.txt)):

```
Execution failed for task ':compileJava'.
> Not a valid class name: com.example.test.package-info
```
Simply running the task again fixes it, but it's still somewhat annoying. I don't get an error with `micronaut.processing.incremental` set to `false`. I also already tried a few different versions, but all show the same behavior:

* Gradle 7.0 (from wrapper), 6.8.3 (user installation) as well as 7.1 (official docker image)
* Micronaut 2.5 and 2.1.0

### Example Application

- [https://github.com/torbsto/micronaut-issue](https://github.com/torbsto/micronaut-issue) (the application is in the directory `incremental-gradle-compilation`)

**Comment by graemerocher** [comment]
This seems like a bug in Gradle, can you report to https://github.com/gradle/gradle/issues

**Comment by torbsto** [comment]
Issue is fixed with Gradle 7.2
